### PR TITLE
Isis Retina - Dropdown icons fix

### PR DESCRIPTION
**Pull request by ciar4n**
Pull Request for Issue #12500 .
### Summary of Changes

Removes sprite image from dropdown on retina displays.
### Testing Instructions

Navigate to article edit. If using a Retina display you should see triple icons on dropdown selects which is incorrect. If not on a retina display, zoom in a few notches on your browser to replicate the issue.

**Before Patch**
retina1

**After Patch**
retina2
### Documentation Changes Required

None

### CODE REVIEW

**Comment by dgt41** [comment]
@ciar4n nice! 
Tested successfully but can't set the result in tracker

BTW I was screaming to use an override for chosen.css but everybody was against it...

**Comment by zero-24** [comment]
@dgt41 what is wrong in the tracker?

**Comment by PhilETaylor** [comment]
Missing commit SHA? 
screen shot 2016-10-23 at 16 33 36

**Comment by PhilETaylor** [comment]
I have tested this and it worked... 

screen shot 2016-10-23 at 16 31 11

**Comment by zero-24** [comment]
@mbabker can you take a look into the commit aha thing?

**Comment by mbabker** [comment]
I can't say I like this fix but if it fixes it then whatever.

Tracker's fixed.

**Comment by dgt41** [comment]
I have tested this item :white_check_mark: successfully on 3186e0b0d296be98414a1ccdf834cc72f3102cbf<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12528">issues.joomla.org/tracker/joomla-cms/12528</a>.</sub>

**Comment by dgt41** [comment]
RTC then

**Comment by dgt41** [comment]
Actually this is kinda wrong as we remove retina support from default chosen.css. I still think that an override is the right approach here!

**Comment by PhilETaylor** [comment]
I have tested this item :white_check_mark: successfully on 3186e0b0d296be98414a1ccdf834cc72f3102cbf<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12528">issues.joomla.org/tracker/joomla-cms/12528</a>.</sub>

**Comment by PhilETaylor** [comment]
Do we have retina support everywhere else though?

**Comment by dgt41** [comment]
@PhilETaylor this is a file from an external repo, we shouldn't mess with it in such way. Override is a clean approach

**Comment by PhilETaylor** [comment]
ah ok

**Comment by ciar4n** [comment]
These fields now use an icon font set which removes the need for a 2x image as the icons are now fully scalable. 

@dgt41 I could override this CSS (instead of removing) in the template.css but that is just doubling up on un-needed CSS?

@mbabker Is your concern about the clarity on retina displays? This is no longer an issue with the font icons.

**Comment by dgt41** [comment]
Also the file was already created and at some point it was decided to not include an override...
the file is here: https://github.com/joomla/joomla-cms/blob/381105ff2467abc3bddf77e7a2752d737c46f96f/administrator/templates/isis/css/jui/chosen.css

**Comment by ciar4n** [comment]
@dgt41 The chosen.css is now overridden within the Isis template.css rather than creating a separate chosen.css for the template. This was to allow for an easier upgrade of the chosen.css if needs be in the future.

**Comment by ciar4n** [comment]
@PhilETaylor Yes. As all chosen icons are now font based ensuring retina support and removes the need for 2x sprites.

**Comment by mbabker** [comment]
> @mbabker Is your concern about the clarity on retina displays? This is no longer an issue with the font icons.

At first glance it just seems like we're removing retina support.  Admittedly my eyes aren't good enough to know what on earth the difference is for retina versus non-retina so it may be a non-issue.

**Comment by dgt41** [comment]
@ciar4n just to clarify my point here:
you've been asked to do the override **inside** the template.css and you delivered that, I am not blaming you here. All I am pointing is that, that decision **was wrong** since we end up with bug and finally with the removal of the retina support for anyone that doesn't override the default `chosen.css`

**Comment by ciar4n** [comment]
A little explanation for those that are wondering..

**3.6.x**
Select fields (chosen.css) used an image for it's icons. As retina displays have a higher pixel density, a second larger image was required to avoid the icon looking pixelated compared to other vector based elements (text, divs etc) on the screen. This is the CSS that has been removed with this PR.

**3.7.x**
Select fields now use a font set for it's icons since #11832. As the icons are now vector based, they will resize without any loss of quality. There is no longer a need to define a different icon for each screen size. The reason this issue presented a number of icons was both the font based (retina ready) and the image based (retina ready) icon was loading. This PR removes the image based icon.

**Conclusion**
Basically the bottom line is these fields still completely supports retina displays, in fact more so than previously as more high resolution/density displays become available.

@dgt41 It is true that this CSS will also have to be removed from any further updates to chosen.css. Something that may be forgotten in the future. Guess we'll have to crossed that hurdle when we get to it! :)

**Comment by brianteeman** [comment]
@dgt41 please clear your browser cache

**Comment by dgt41** [comment]
@brianteeman ???

**Comment by brianteeman** [comment]
No idea why that message just came through. I wasn't even online 3hours ago

**Comment by PhilETaylor** [comment]
masses of @brianteeman emails came through to GH in one sweep about 3 hours ago - seems a GH issue

**Comment by brianteeman** [comment]
Weird

**Comment by andrepereiradasilva** [comment]
mantainers please add 3.7.0 label here

**Comment by demis-palma** [comment]
@ciar4n 
> Guess we'll have to cross that hurdle when we get to it!

The time has come. :smile: I have a regression due to a fix overwrite. See https://github.com/joomla/joomla-cms/pull/16552 and kindly confirm.

PS: @dgt41 was terribly right.

**Comment by demis-palma** [comment]
@ciar4n
> Select fields now use a font set for it's icons since

1. Why the file media/jui/css/chosen-sprite@2x.png is still there?
2. And since it contains the same icons as chosen-sprite.png why this one is still there as well?
3. And why media/jui/css/chosen.css still contains a lot of references to 'chosen-sprite.png'?

**Comment by zero-24** [comment]
@demis-palma can you do a new issue with that questions? Else this would get lost fast ;)

**Comment by demis-palma** [comment]
It's actually not an issue, a the moment it's a genuine question. I guess that there is a reason behind that, but I still don't understand what it is.

**Comment by ciar4n** [comment]
@demis-palma 

If I remember correctly media/jui/css/chosen.css is untouched and is instead overridden by the template CSS where needed. So reference to the chosen-sprite is overridden by the template which replaces the sprite with a font set. Discussion on this can be found here... https://github.com/joomla/joomla-cms/pull/11832#issuecomment-244128681
